Document/0x06b-Basic-Security-Testing.md

```diff
@@ -786,42 +757,298 @@ Possible values for the property [UIDeviceFamily](https://developer.apple.com/li
+The most relevant items are:
 
-There is much more information provided that you should explore, that might be helpful for you.
+- `Info.plist` contains configuration information for the application, such as its bundle ID, version number, and display name.
+- resources such as images/icons, localized content (`<language>.lproj`), text files, audio files, etc.
+- `*.nib` files store the user interfaces of iOS app.
+- `_CodeSignature/` contains a plist file with a signature over all files in the bundle.
+- `Frameworks/` contains the app native libraries as .dylib or .framework files.
+- `PlugIns/` may contain app extensions as .appex files (not present in the example).
+- `iGoat-Swift` is the app binary containing the app’s code. Its name is the same as the bundle's name minus the .app extension.
 
-###### Objection
+###### The Info.plist File
+
+The information property list or `Info.plist` (named by convention) is main source of information for an iOS app. It consists of a structured file containing key-value pairs describing essential configuration information about the app. Actually, all bundled executables (app extensions, frameworks and apps) are expected to have an `Info.plist` file. You can find all possible key in the [Apple Developer Documentation](https://developer.apple.com/documentation/bundleresources/information_property_list?language=objc "Information Property List").
+
+The file format might be XML or binary (bplist). You can convert it to XML format with one simple command:
+
+- On macOS with `plutil`, which is a tool that comes natively with macOS 10.2 and above versions (no official online documentation is currently available):
+
+  ```bash
+  $ plutil -convert xml1 Info.plist
+  ```
+
+- On Linux:
+
+  ```bash
+  $ apt install libplist-utils
+  $ plistutil -i Info.plist -o Info_xml.plist
+  ```
+
+Here's a non-exhaustive list of some info and the corresponding keywords that you can easily search for in the `Info.plist` file by just inspecting the file or by using `grep -i <keyword> Info.plist`:
+
+- App permissions Purpose Strings: `UsageDescription` (see "iOS Platform APIs")
+- Custom URL schemes: `CFBundleURLTypes` (see "iOS Platform APIs")
+- Exported/imported *custom document types*: `UTExportedTypeDeclarations`/`UTImportedTypeDeclarations` (see "iOS Platform APIs")
+- App Transport Security (ATS) configuration: `NSAppTransportSecurity` (see "iOS Network APIs")
+
+Please refer to the mentioned chapters to learn more about how to test each of these points.
+
+###### App Binary
+
+iOS app binaries are fat binaries (they can be deployed on all devices 32- and 64-bit). In contrast to Android, where you can actually decompile the app binary to Java code, the iOS app binaries can only be disassembled.
+
+Refer to the chapter "Reverse Engineering and Tampering on iOS" for more details.
+
+###### Native Libraries
+
+iOS native libraries are known as Frameworks. You can inspect them by listing the `Frameworks` folder in the IPA:
```

Comment:
I think you refer to 0x6i? I've added this sentence:

Please note that this might not be the complete list of third party libraries being used by the app as some can be part of the source code, meaning that they'll be compiled in the app binary and therefore cannot be found as standalone libraries or Frameworks in the `Frameworks` folder.